### PR TITLE
Sanity Source

### PR DESCRIPTION
**Pull request by pingevt**
I added the Sanity Source. I'm not sure where to add the examples for source params.

Does not include:

- [ ] tests
- [ ] source documentation

### CODE REVIEW

File: packages/content/lib/content-sources/sanity-source.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+			// join the paragraphs leaving split by two linebreaks
+			.join('\n\n')
+	}
+
+	/**
+	 *
+	 * @param {*} config
+	 * @returns {SanityOptions}
+	 */
+	static _assembleConfig(config) {
```
Comment (benjaminbojko) [inline packages/content/lib/content-sources/sanity-source.js:265]: Fix indentation

File: packages/content/lib/content-sources/sanity-source.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+  /**
+   * Recursively fetches content using the Sanity client.
+   *
+   * @param {string} query
+	 * @param {string} jwt The JSON web token generated by Sanity
+   * @param {ContentResult} result
+   * @param {Object} params
+   * @returns {Promise<Object>} Object with an 'entries' and an 'assets' array.
+   */
+  async _fetchPages(
```
Comment (benjaminbojko) [inline packages/content/lib/content-sources/sanity-source.js:151]: I haven't looked much into Sanity's SDK, but how close is it to a general GraphQL API? Or is it using Sanity's own GROQ?

File: packages/content/lib/content-sources/sanity-source.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+		apiVersion = "v2021-10-21",
+		projectid = undefined,
+		dataset = "production",
+		apiToken = undefined,
+		useCdn = false,
+    baseUrl = undefined,
+		queries = [],
+		textConverters = [],
+		limit = 100,
+		maxNumPages = -1,
```
Comment (benjaminbojko) [inline packages/content/lib/content-sources/sanity-source.js:31]: Have you tested various pagination scenarios? Speaking from my own experience that's often an area that breaks much later down the road once you reach a significant amount of content 😅

**Comment by benjaminbojko** [comment]
In regards to testing, you could add a test source here: https://github.com/bluecadet/launchpad/tree/develop/packages/content/test

**Comment by pingevt** [comment]
@benjaminbojko - 

1. My gut response is no... Sanity has its structured data based off of "[Portable Text](https://github.com/portabletext/portabletext)" Which I think is a sanity thing anyways. So if data is coming through as Portable text, then yes these transforms could work in other sources. Also, this doesn't account for custom types of blocks in Portable text. We could custom define a whole structured, but without that definition, theres no way to transform it into HTML or plain text or md, etc. So, there are a few gaps here we could discuss. Also, I'm just using the npm packages Sanity provides for those transforms. I didn't code any of the transform work. The plain text one was a copy/paste from Sanity docs.
2. So Sanity uses GROQ, instead of GraphQL (I don't know why - it looks very similar). I think the source created here is a bare bones doc type => json file. Grab everything typeof implementation. So even here we are getting drafts in the data, but I am filtering them out in Unity. I think we could expand this to include custom queries, but I would need to dive in a bit further. But I do think it would be hard to traverse the data and automatically create data.

Working through your other comments today

File: packages/content/lib/launchpad-content.js [hunk 4]
```diff
@@ -46,19 +48,19 @@ export class LaunchpadContent {
       return Promise.reject(error);
     }
   }
-  
+
```
[the remark this answers is not in the document]
Comment (pingevt) [reply packages/content/lib/launchpad-content.js:51]: @benjaminbojko for sure.. I don't have anything setup for JS to prettify. This specific line, I do have a VSCode plugin that removes extra whitespace from lines. We could setup a prettifier. Do we have our standard set somewhere?

File: packages/content/lib/content-sources/sanity-source.js [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+ * Options for SanitySource
+ */
+export class SanityOptions extends SourceOptions {
+	constructor({
+		apiVersion = "v2021-10-21",
+		projectid = undefined,
+		dataset = "production",
+		apiToken = undefined,
+		useCdn = false,
+    baseUrl = undefined,
```
[the remark this answers is not in the document]
Comment (pingevt) [reply packages/content/lib/content-sources/sanity-source.js:27]: @benjaminbojko - are you team tabs or spaces?

File: packages/content/lib/content-sources/sanity-source.js (hunk 2 above)
Comment (pingevt) [reply packages/content/lib/content-sources/sanity-source.js:151]: Its GROQ. I think we could have this setup as just generic grab data... but then also add custom queries.

**Comment by benjaminbojko** [comment]
🙌

**Pull request merged**